control/plugin/rpc/plugin.proto

```diff
@@ -42,9 +42,36 @@ service Publisher {
     rpc GetConfigPolicy(Empty) returns (GetConfigPolicyReply) {}
 }
 
-message Empty{
+service StreamCollector {
+	rpc StreamMetrics(stream CollectArg) returns (stream CollectReply) {}
+    rpc GetMetricTypes(GetMetricTypesArg) returns (MetricsReply) {}
+    rpc Ping(Empty) returns (ErrReply) {}
+    rpc Kill(KillArg) returns (ErrReply) {}
+    rpc GetConfigPolicy(Empty) returns (GetConfigPolicyReply) {}
 }
 
+// Request that can be passed a stream collector
+message CollectArg{
+	// Request these metrics to be collected on the plugins schedule
+	MetricsArg Metrics_Arg = 1;
+	// Set minimum collection duration --duration in ns
+	int64 MaxCollectDuration = 2;
+	// Set max number of metrics to buffer before forcing send
+	// 0 means no forced send
```

Comment:
@jcooklin: Are you proposing an alternate behavior or trying to discuss possibilities with the current behavior? I was viewing the default state as giving the plugin no restrictions so MaxCollectDuration/MaxMetricsBuffer being set to 0 means the plugin can make it's own determination. Maybe this isn't the intended behavior?

@candysmurf: I think they cover different but partially overlapping use cases.